File: rental-backend/src/main/java/com/homeflex/features/property/dto/response/BookingDto.java
package com.homeflex.features.property.dto.response;

import com.homeflex.core.dto.response.UserDto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

public record BookingDto(
        UUID id,
        PropertyDto property,
        UserDto tenant,
        String bookingType,
        LocalDateTime requestedDate,
        LocalDate startDate,
        LocalDate endDate,
        String status,
        String message,
        Integer numberOfOccupants,
        BigDecimal totalPrice,
        BigDecimal platformFee,
        BigDecimal cleaningFee,
        BigDecimal taxAmount,
        String stripePaymentIntentId,
        String stripeClientSecret,
        String paymentStatus,
        String paymentFailureReason,
        String landlordResponse,
        LocalDateTime respondedAt,
        LocalDate proposedStartDate,
        LocalDate proposedEndDate,
        String modificationReason,
        LocalDateTime createdAt,
        // Hotel room fields
        UUID roomTypeId,
        String roomTypeName,
        Integer numberOfRooms,
        LocalDateTime paymentConfirmedAt,
        LocalDateTime escrowReleasedAt
) {}

File: rental-app-flutter/lib/features/bookings/models/booking_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import '../../properties/models/property_dto.dart';
import '../../auth/models/user_dto.dart';

part 'booking_dto.freezed.dart';
part 'booking_dto.g.dart';

@freezed
abstract class BookingDto with _$BookingDto {
  const factory BookingDto({
    required String id,
    PropertyDto? property,
    UserDto? tenant,
    required String bookingType,
    String? requestedDate,
    String? startDate,
    String? endDate,
    required String status,
    String? message,
    int? numberOfOccupants,
    double? totalPrice,
    double? platformFee,
    String? stripePaymentIntentId,
    String? paymentConfirmedAt,
    String? escrowReleasedAt,
    String? landlordResponse,
    String? respondedAt,
    required String createdAt,
  }) = _BookingDto;

  factory BookingDto.fromJson(Map<String, dynamic> json) => _$BookingDtoFromJson(json);
}
